crates/base/src/deno_runtime.rs

```diff
@@ -435,15 +527,44 @@ impl DenoRuntime {
             .load_main_module(&main_module_url, mod_code)
             .await?;
 
+        if is_user_worker {
+            drop(rt::SUPERVISOR_RT.spawn({
+                let drop_token = mem_check_state.drop_token.clone();
+                let waker = mem_check_state.waker.clone();
+
+                async move {
+                    // TODO(Nyannyacha): Should we introduce exponential
```

Comment:
yeah, that'd be neat! Can be a follow up PR?

Reply:
@laktek 
Planning to submit a PR in the next iteration.
In my own degradation experiments (see slack), I didn't find any significant degradation, so I just left it as a comment in this PR.